### PR TITLE
Add a CONTRIBUTING.md file

### PR DESCRIPTION
**Pull request by drigz**
This should help new contributors test their changes correctly.

It is based on
https://github.com/thoughtbot/factory_bot_rails/blob/master/CONTRIBUTING.md.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+Make your change. Add tests for your change. Make the tests pass:
+
+    bazel test //...
+
+If your change affects the generated `workspace.bzl`, regenerate it:
+
+    bazel build src/scala/com/github/johnynek/bazel_deps:parseproject_deploy.jar
+    ./gen_maven_deps.sh generate -r $PWD \
+        -s 3rdparty/workspace.bzl -d dependencies.yaml
```
Comment (jin) [inline CONTRIBUTING.md]: you can now run 

`bazel run //:parse -- generate -r $PWD -s 3rdparty/workspace.bzl -d dependencies.yml`

Comment (johnynek) [reply CONTRIBUTING.md]: Yes, can we make the above change? It makes it faster to run since you don’t need to make the deploy jar.

Comment (drigz) [reply CONTRIBUTING.md]: Done - that is better.

**Comment by johnynek** [comment]
Thank you!

**Pull request merged**